### PR TITLE
API Generator: Add support for filtering many-to-many-relations by id

### PR DESCRIPTION
**Pull request by nsams**

### CODE REVIEW

File: packages/api/cms-api/src/common/filter/mikro-orm.ts [hunk 1]
```diff
@@ -95,6 +96,12 @@ export function filterToMikroOrmQuery(
         if (filterProperty.isAnyOf !== undefined) {
             ret.$in = filterProperty.isAnyOf;
         }
+    } else if (filterProperty instanceof ManyToManyFilter) {
+        if (filterProperty.contains !== undefined) {
+            ret.id = {
```
Comment (nsams) [inline packages/api/cms-api/src/common/filter/mikro-orm.ts:101]: hardcoding id here means the primary key of the related entity must be `id`. Far from ideal, but we currently don't have any entity metadata in this function.

Comment (johnnyomair) [reply packages/api/cms-api/src/common/filter/mikro-orm.ts:101]: Should we add an options argument to override the key? Is this even necessary?

File: .changeset/unlucky-points-matter.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-api": minor
+---
+
+CRUD Generator: Add support for filtering ManyToMany relations by id
```
Comment (johnnyomair) [inline .changeset/unlucky-points-matter.md:5]:
```suggestion
API Generator: Add support for filtering many-to-many-relations by id
```

**Review by johnnyomair — COMMENTED** [review]
Conflicts 🙄

File: packages/api/cms-api/src/common/filter/mikro-orm.ts (hunk 1 above)
Comment (nsams) [reply packages/api/cms-api/src/common/filter/mikro-orm.ts:102]: in practice it's called id in maybe 99%

Comment (johnnyomair) [reply packages/api/cms-api/src/common/filter/mikro-orm.ts:102]: Then we add support to override it if we ever need it.

File: packages/api/cms-api/src/generator/generate-crud.ts [hunk 3]
```diff
@@ -205,7 +206,8 @@ function generateFilterDto({ generatorOptions, metadata }: { generatorOptions: C
                     @Type(() => ${enumName}EnumFilter)
                     ${prop.name}?: ${enumName}EnumFilter;
                     `;
-                } else if (prop.type === "string" || prop.type === "text") {
+                } else if (prop.type 
+                           "string" || prop.type === "text") {
```
Comment (johnnyomair) [inline packages/api/cms-api/src/generator/generate-crud.ts:210]: This looks like a merge mistake to me

**Review by johnnyomair — CHANGES_REQUESTED** [review]

**Review by johnnyomair — DISMISSED** [review]

**Comment by johnnyomair** [comment]
@nsams the lint workflow fails

**Pull request merged**